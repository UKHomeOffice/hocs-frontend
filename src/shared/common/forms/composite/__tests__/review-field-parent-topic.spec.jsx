import React from 'react';
import ParentTopicReviewField from '../review-field-parent-topic.jsx';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';

const parentTopicChoices = [
    {
        'label': 'Parent Topic 1',
        'value': 'parent_1',
        'options': [
            {
                'label': 'Topic 11',
                'value': 'topic_11',
                'active': true
            },
            {
                'label': 'Topic 12',
                'value': 'topic_12',
                'active': true
            }
        ]
    },
    {
        'label': 'Parent Topic 2',
        'value': 'parent_2',
        'options': [
            {
                'label': 'Topic 21',
                'value': 'topic_21',
                'active': true
            },
            {
                'label': 'Topic 22',
                'value': 'topic_22',
                'active': true
            }
        ]
    }
];



describe('Review Field Parent Topic',  () => {

    it('Should Render Topic in review field', () => {
        const name = 'TEST NAME';
        const type = 'TEST TYPE';
        const value = 'topic_21';
        const page = {
            caseId: 'case_id',
            stage: 'stage_id'
        };

        const child = {
            component: 'type-ahead',
            props: {
                choices: parentTopicChoices
            }
        };

        const wrapper = render(<ParentTopicReviewField
            name={name}
            child={child}
            type={type}
            value={value}
            page={page}/>
        );

        expect(wrapper).toBeDefined();
        expect(wrapper).toMatchSnapshot();
    });
});
