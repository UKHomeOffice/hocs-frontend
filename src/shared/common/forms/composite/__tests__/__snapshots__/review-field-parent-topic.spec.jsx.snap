// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review Field Parent Topic Should Render Topic in review field 1`] = `
<div
  class="govuk-grid-row grey-border-bottom"
>
  <div
    class="govuk-grid-column-one-half"
  >
    <span
      class="govuk-label govuk-label--s"
    />
  </div>
  <div
    class="govuk-grid-column-one-third govuk-body review-field-column"
    id="TEST NAME"
    name="TEST NAME"
    type="TEST TYPE"
  >
    Topic 21
  </div>
  <div
    class="grid-column-one-sixth-right-align govuk-body review-field-column"
  >
    <a
      class="govuk-body govuk-link change-link"
      href=""
    >
      Change 
      <span
        class="govuk-visually-hidden"
      />
    </a>
  </div>
</div>
`;
